### PR TITLE
hubble: Add "syn-only" option to flows-to-world metric

### PR DESCRIPTION
**Pull request by michi-covalent**
2 commits:

- add `syn-only` option.
- implement `Status()` method so that it prints option flags. for example:
  `level=info msg="Configured metrics plugin" name=flows-to-world status="syn-only,destination=dns|reserved-identity,source=namespace" subsys=hubble`

```release-note
hubble: Add "syn-only" option to flows-to-world metric
```

### CODE REVIEW

File: pkg/hubble/metrics/flows-to-world/handler.go [hunk 1]
```diff
@@ -85,6 +98,12 @@ func (h *flowsToWorldHandler) ProcessFlow(_ context.Context, flow *flowpb.Flow)
 	if flow.GetVerdict() != flowpb.Verdict_DROPPED && isReply {
 		return nil
 	}
+
+	// if "syn-only" option is set, only count non-reply SYN packets for TCP.
+	if h.synOnly && l4.GetTCP() != nil && l4.GetTCP().GetFlags() != nil && (!l4.GetTCP().GetFlags().SYN || isReply) {
```
Comment (chancez) [inline pkg/hubble/metrics/flows-to-world/handler.go]: You can avoid the `nil` checks and just use `l4.GetTCP().GetFlags().GetSYN()`
eg:
```
if h.synOnly && (!l4.GetTCP().GetFlags().GetSYN() || isReply) {
```

Comment (michi-covalent) [reply pkg/hubble/metrics/flows-to-world/handler.go]: magical 💯

**Comment by michi-covalent** [comment]
- test-1.25-net-next: known flake: https://github.com/cilium/cilium/issues/21120

**Comment by michi-covalent** [comment]
/test-1.25-net-next

**Review by gandro — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Review by chancez — APPROVED** [review]

**Comment by michi-covalent** [comment]
not sure why sig-policy team was added to review, this PR only touches hubble metrics. i'm marking it as ready-to-merge.

**Pull request merged**